src/main/java/soot/jimple/validation/FieldRefValidator.java

```diff
@@ -66,26 +66,25 @@ public void validate(Body body, List<ValidationException> exceptions) {
         try {
           SootField field = v.getField();
           if (field == null) {
-            exceptions.add(new UnitValidationException(unit, body, "Resolved field is null: " + fr.toString()));
+            exceptions.add(new UnitValidationException(u, body, "Resolved field is null: " + fr.toString()));
           } else if (!field.isStatic() && !field.isPhantom()) {
-            exceptions
-                .add(new UnitValidationException(unit, body, "Trying to get a static field which is non-static: " + v));
+            exceptions.add(new UnitValidationException(u, body, "Used StaticFieldRef for a non-static field: " + v));
           }
         } catch (ResolutionFailedException e) {
-          exceptions.add(new UnitValidationException(unit, body, "Trying to get a static field which is non-static: " + v));
+          exceptions.add(new UnitValidationException(u, body, "Unable to resolve FieldRef " + v + ": " + e.getMessage()));
         }
       } else if (fr instanceof InstanceFieldRef) {
         InstanceFieldRef v = (InstanceFieldRef) fr;
 
         try {
           SootField field = v.getField();
           if (field == null) {
-            exceptions.add(new UnitValidationException(unit, body, "Resolved field is null: " + fr.toString()));
+            exceptions.add(new UnitValidationException(u, body, "Resolved field is null: " + fr.toString()));
           } else if (field.isStatic() && !field.isPhantom()) {
-            exceptions.add(new UnitValidationException(unit, body, "Trying to get an instance field which is static: " + v));
+            exceptions.add(new UnitValidationException(u, body, "Used InstanceFieldRef for a static field: " + v));
           }
         } catch (ResolutionFailedException e) {
-          exceptions.add(new UnitValidationException(unit, body, "Trying to get an instance field which is static: " + v));
+          exceptions.add(new UnitValidationException(u, body, "Unable to resolve FieldRef " + v + ": " + e.getMessage()));
```

Comment:
sometimes you append e.getMessage() to the message, sometimes you don't. The original messages contain what kind (static, or instance) of fields soot was trying to resolve. I don't see improvement in the messages. the variable unit was also better than u.

Reply:
If you take a closer look, I append ```e.getMessage()``` in those cases where an exception ```e``` was caught. In the other cases, there is no exception to append.
Your second sentence I don't understand what you are trying to say. Are you talking about the 2 cases where I totally changed the text String because it was saying the incorrect thing (it seems to have been just copy pasted from an earlier line)? 
Regarding your final comment, I simply changed ```unit``` to ```u``` because it allowed the autoformatting rules (specifically the maximum line length rule) to not split the lines so much and the method was overall small enough that I figured the name ```u``` for the Unit would be obvious enough. But I can change that back if you like as it was simply a matter of preference and formatting.